projects/sandbox/train/tests/test_validation.py

```diff
@@ -128,7 +129,7 @@ def test_shift_background(self, background):
 
     def test_iter_shift(self, background, psd_estimator, whitener):
         mock = Mock()
-        mock.background = background
+        mock.current_segment = background[0]
         mock.sample_rate = 8
```

Comment:
I think you need a `mock.current_duration = ` here and in `test_steps_for_shift`